nil pointer on Unmarshal Tx

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xd0 pc=0x2c288cd]

goroutine 12816418 [running]:
go.vocdoni.io/dvote/vochain/transaction/vochaintx.(*Tx).Unmarshal(0xc01c6335e0, {0x0, 0x0, 0x0}, {0xc002930060, 0xe})
	go.vocdoni.io/dvote/vochain/transaction/vochaintx/vochaintx.go:39 +0x20d
go.vocdoni.io/dvote/vochain.(*BaseApplication).CheckTx(0xc0044d6420, {0x40?, 0x36c0a00?}, 0xc01c62c4a0)
	go.vocdoni.io/dvote/vochain/cometbft.go:196 +0x44a
github.com/cometbft/cometbft/abci/client.(*localClient).CheckTxAsync(0xc0058107e0, {0x4513f80?, 0x5ff4ba0?}, 0xc01c62c4a0)
	github.com/cometbft/cometbft@v0.38.0/abci/client/local_client.go:51 +0xdf
github.com/cometbft/cometbft/proxy.(*appConnMempool).CheckTxAsync(0xc0055cf710, {0x4513f80, 0x5ff4ba0}, 0x60?)
	github.com/cometbft/cometbft@v0.38.0/proxy/app_conn.go:147 +0x170
github.com/cometbft/cometbft/mempool.(*CListMempool).CheckTx(0xc001b7e300, {0x0, 0x0, 0x0}, 0xc0019886a8, {0xd4c0?, {0x0?, 0x1c93397?}})
	github.com/cometbft/cometbft@v0.38.0/mempool/clist_mempool.go:268 +0x20b
github.com/cometbft/cometbft/rpc/core.(*Environment).BroadcastTxSync(0xc0004b2fc0, 0xc005285100, {0x0, 0x0, 0x0})
	github.com/cometbft/cometbft@v0.38.0/rpc/core/mempool.go:35 +0xec
github.com/cometbft/cometbft/rpc/client/local.(*Local).BroadcastTxSync(...)
	github.com/cometbft/cometbft@v0.38.0/rpc/client/local/local.go:98
go.vocdoni.io/dvote/vochain.(*BaseApplication).SetDefaultMethods.func4({0x0?, 0x0, 0x0?})
	go.vocdoni.io/dvote/vochain/appsetup.go:64 +0x4e
go.vocdoni.io/dvote/vochain.(*BaseApplication).SendTx(0xc001dbadb0?, {0x0?, 0x4538128?, 0xc00e17d4c0?})
	go.vocdoni.io/dvote/vochain/app.go:325 +0x2a
go.vocdoni.io/dvote/api.(*API).sendTx(0xc017b0e300?, {0x0?, 0x200?, 0x35e8620?})
	go.vocdoni.io/dvote/api/helpers.go:48 +0x2b
go.vocdoni.io/dvote/api.(*API).accountSetHandler(0xc00dc2c4d0, 0xc0107f68a0, 0xc00059ccf0?)
	go.vocdoni.io/dvote/api/accounts.go:230 +0x2a6
go.vocdoni.io/dvote/httprouter/apirest.(*API).RegisterMethod.func1({{0x35ed6e0, 0xc0107f68a0}, {0xc14febf1307276f2, 0x7baed980287, 0x5fbd0a0}, {0xc0107f68e0, 0x2, 0x2}, 0xc01c62c480})
	go.vocdoni.io/dvote/httprouter/apirest/apirest.go:229 +0x6b
created by go.vocdoni.io/dvote/httprouter.(*HTTProuter).AddPublicHandler.(*HTTProuter).routerHandler.func1 in goroutine 12816372
	go.vocdoni.io/dvote/httprouter/httprouter.go:283 +0x3c5
```

--- a/api/accounts.go
+++ b/api/accounts.go
@@ -180,6 +180,10 @@ func (a *API) accountHandler(_ *apirest.APIdata, ctx *httprouter.HTTPContext) er
 //	@Router					/accounts [post]
 func (a *API) accountSetHandler(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
 	req := &AccountSet{}
+	if msg == nil || msg.Data == nil {
+		return ErrCantParseDataAsJSON
+	}
+
 	if err := json.Unmarshal(msg.Data, req); err != nil {
 		return err
 	}
@@ -187,11 +191,18 @@ func (a *API) accountSetHandler(msg *apirest.APIdata, ctx *httprouter.HTTPContex
 	// check if the transaction is of the correct type and extract metadata URI
 	metadataURI, err := func() (string, error) {
 		stx := &models.SignedTx{}
+		if req.TxPayload == nil {
+			return "", ErrUnmarshalingServerProto
+		}
 		if err := proto.Unmarshal(req.TxPayload, stx); err != nil {
 			return "", err
 		}
 		tx := &models.Tx{}
-		if err := proto.Unmarshal(stx.GetTx(), tx); err != nil {
+		gotTx := stx.GetTx()
+		if gotTx == nil {
+			return "", ErrUnmarshalingServerProto
+		}
+		if err := proto.Unmarshal(gotTx, tx); err != nil {
 			return "", err
 		}
 		if np := tx.GetSetAccount(); np != nil {

--- a/vochain/cometbft.go
+++ b/vochain/cometbft.go
@@ -168,6 +168,12 @@ func (app *BaseApplication) InitChain(_ context.Context,
 // CheckTx unmarshals req.Tx and checks its validity
 func (app *BaseApplication) CheckTx(_ context.Context,
 	req *abcitypes.RequestCheckTx) (*abcitypes.ResponseCheckTx, error) {
+	if req == nil || req.Tx == nil {
+		return &abcitypes.ResponseCheckTx{
+			Code: 1,
+			Data: []byte("nil request or tx"),
+		}, fmt.Errorf("nil request or tx")
+	}
 	txReference := vochaintx.TxKey(req.Tx)
 	ref, ok := app.txReferences.Load(txReference)
 	if !ok {
@@ -219,6 +225,9 @@ func (app *BaseApplication) FinalizeBlock(_ context.Context,
 	req *abcitypes.RequestFinalizeBlock) (*abcitypes.ResponseFinalizeBlock, error) {
 	app.prepareProposalLock.Lock()
 	defer app.prepareProposalLock.Unlock()
+	if req == nil {
+		return nil, fmt.Errorf("nil request")
+	}
 	start := time.Now()
 	height := uint32(req.GetHeight())
 
@@ -327,6 +336,9 @@ func (app *BaseApplication) PrepareProposal(ctx context.Context,
 	req *abcitypes.RequestPrepareProposal) (*abcitypes.ResponsePrepareProposal, error) {
 	app.prepareProposalLock.Lock()
 	defer app.prepareProposalLock.Unlock()
+	if req == nil {
+		return nil, fmt.Errorf("nil request")
+	}
 	startTime := time.Now()
 
 	type txInfo struct {
@@ -431,6 +443,9 @@ func (app *BaseApplication) ProcessProposal(_ context.Context,
 	req *abcitypes.RequestProcessProposal) (*abcitypes.ResponseProcessProposal, error) {
 	app.prepareProposalLock.Lock()
 	defer app.prepareProposalLock.Unlock()
+	if req == nil {
+		return nil, fmt.Errorf("nil request")
+	}
 	// Check if the node is a validator, if not, just accept the proposal and return (nothing to say)
 	validator, err := app.State.Validator(app.NodeAddress, true)
 	if err != nil {

--- a/vochain/transaction/vochaintx/vochaintx.go
+++ b/vochain/transaction/vochaintx/vochaintx.go
@@ -36,7 +36,15 @@ func (tx *Tx) Unmarshal(content []byte, chainID string) error {
 	if err != nil {
 		return fmt.Errorf("failed to unmarshal transaction: %w", err)
 	}
-	tx.TxModelType = string(tx.Tx.ProtoReflect().WhichOneof(tx.Tx.ProtoReflect().Descriptor().Oneofs().Get(0)).Name())
+	txReflectDescriptor := tx.Tx.ProtoReflect().Descriptor().Oneofs().Get(0)
+	if txReflectDescriptor == nil {
+		return fmt.Errorf("failed to determine transaction type")
+	}
+	whichOneTxModelType := tx.Tx.ProtoReflect().WhichOneof(txReflectDescriptor)
+	if whichOneTxModelType == nil {
+		return fmt.Errorf("failed to determine transaction type")
+	}
+	tx.TxModelType = string(whichOneTxModelType.Name())
 	tx.Signature = stx.GetSignature()
 	tx.TxID = TxKey(content)
 	return nil

